tests/integration/targets/ec2_vpc_dhcp_option/tasks/main.yml

```diff
@@ -0,0 +1,893 @@
+        - dhcp_options.new_options['ntp-servers'] | sort == ['10.0.0.2', '10.0.1.2']
+        - new_dhcp_options_id == dhcp_options.dhcp_options_id
+        - dhcp_options_info.dhcp_options[0].tags is defined
+        - dhcp_options_info.dhcp_options[0].tags.keys() | length == 2
+        - dhcp_options_info.dhcp_options[0].tags['collection'] == 'amazon.aws'
+        - dhcp_options_info.dhcp_options[0].tags['created_by'] == 'ansible-test'
+
+  # FIXME: the additional tag is silently ignored and no change is made
+  - name: add a tag without using dhcp_options_id
+    ec2_vpc_dhcp_option:
+      state: present
+      vpc_id: "{{ vpc_id }}"
+      inherit_existing: False
+      delete_old: True
+      ntp_servers:
+          - 10.0.0.2
+          - 10.0.1.2
+      netbios_name_servers:
+          - 10.0.0.1
+          - 10.0.1.1
+      tags:
+        CreatedBy: ansible-test
+        Collection: amazon.aws
+        another: tag
+    register: dhcp_options
+
+  - name: check if the expected tags are associated
+    ec2_vpc_dhcp_option_info:
+      dhcp_options_ids: ["{{ dhcp_options.dhcp_options_id }}"]
+    register: dhcp_options_info
+
+  - assert:
+      that:
+         #- dhcp_options.changed
+        - dhcp_options.new_options.keys() | list | sort == ['netbios-name-servers', 'ntp-servers']
+        - dhcp_options.new_options['netbios-name-servers'] | sort == ['10.0.0.1', '10.0.1.1']
+        - dhcp_options.new_options['ntp-servers'] | sort == ['10.0.0.2', '10.0.1.2']
+        - new_dhcp_options_id == dhcp_options.dhcp_options_id
+        - dhcp_options_info.dhcp_options[0].tags is defined
+        - dhcp_options_info.dhcp_options[0].tags.keys() | length == 2
+        #- dhcp_options_info.dhcp_options[0].tags.keys() | length == 3
+        - dhcp_options_info.dhcp_options[0].tags['collection'] == 'amazon.aws'
+        - dhcp_options_info.dhcp_options[0].tags['created_by'] == 'ansible-test'
+
+  # FIXME: another check_mode traceback
+  #- name: add and removing tags (check mode)
+  #  ec2_vpc_dhcp_option:
+  #    dhcp_options_id: "{{ dhcp_options.dhcp_options_id }}"
+  #    state: present
+  #    vpc_id: "{{ vpc_id }}"
+  #    inherit_existing: False
+  #    delete_old: True
+  #    ntp_servers:
+  #        - 10.0.0.2
+  #        - 10.0.1.2
+  #    netbios_name_servers:
+  #        - 10.0.0.1
+  #        - 10.0.1.1
+  #    tags:
+  #      AnsibleTest: integration
+  #      Collection: amazon.aws
+  #  register: dhcp_options
+  #  check_mode: true
+
+  #- assert:
+  #    that:
+  #      - dhcp_options.changed
+
+  - name: add and remove tags
+    ec2_vpc_dhcp_option:
+      dhcp_options_id: "{{ dhcp_options.dhcp_options_id }}"
+      state: present
+      vpc_id: "{{ vpc_id }}"
+      inherit_existing: False
+      delete_old: True
+      ntp_servers:
+          - 10.0.0.2
+          - 10.0.1.2
+      netbios_name_servers:
+          - 10.0.0.1
+          - 10.0.1.1
+      tags:
+        AnsibleTest: integration
+        Collection: amazon.aws
```

Comment:
We have a history of accidentally mangling tags (during snake-/Camel- case conversions) in modules.  I'd like to see some additional tag based tests here.

```
tags:
  "lowercase spaced": 'hello cruel world'
  "Title Case": 'Hello Cruel World'
  CamelCase: 'SimpleCamelCase'
  snake_case: 'simple_snake_case'
```

Reply:
I'm not sure about the "Title Case" test, I couldn't find other examples of where we're doing this, but it's in there now. What do you think?

Reply:
Yeah that was me try to be thorough on the test suite I'd started writing.  I think we need to look at being more consistent in general with out tagging tests.  The most common issues are with the other 3, but I don't think adding extra tags would hurt.

Reply:
Sorry not opposed to having more tests. More that I wasn't sure `['title _case']` was a desirable result, though it is what the module is doing.

Reply:
It's not a desirable result, which is why I wanted to test it :)